platforms/core-runtime/daemon-protocol/build.gradle.kts

```diff
@@ -21,16 +21,20 @@ plugins {
 description = "The messages and types sent between client and daemon"
 
 dependencies {
+    api(libs.jsr305)
     api(project(":base-services"))
     api(project(":logging-api"))
     api(project(":serialization"))
     api(project(":logging"))
+    api(project(":core-api"))
```

Comment:
I think the comment below about not depending on core should cover this dependency as well. 

Since many things in `core-api` are implemented in `core`, a dependency on `core-api` is almostly definitely means you likely depend on `core` at runtime

Reply:
Ah yes, it should indeed.

Reply:
Fixed.

I dug a little more into the things that the client is using from core and core-api. It's tangled enough that I think detangling the client from core/core-api should happen in a separate PR (or more).